pydoctor/templatewriter/pages/__init__.py

```diff
@@ -419,3 +511,21 @@ def functionExtras(self, data):
                 )))
         r.extend(super().functionExtras(data))
         return r
+        
+# Type alias for HTML page: to use in type checking. 
+# It's necessary because L{BasePage} doesn't have the same C{__init__} siganture as concrete pages classes. 
```

Comment:
All the listed classes inherit from `CommonPage` and that class is also in the list, so maybe using `CommonPage` instead of `AnyDocPage` would be sufficient?